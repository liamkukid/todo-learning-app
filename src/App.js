import { useState } from 'react';

import { SHOW_ALL } from './Filters';
import Input from './Components/Input';
import List from './Components/List';
import CommandPanel from './Components/CommandPanel';
import style from './App.module.scss';
import useTodosStore from './Store/UseTodosStore';

export default function App() {
  const [filter, setFilter] = useState(SHOW_ALL);
  const [todos, addNewTodo, todoDone, todoRemove, removeCompleted] =
    useTodosStore();

  return (
    <div className={style.app}>
      <div
        style={{
          backgroundImage: `url(${`${process.env.PUBLIC_URL}/work_desk.jpg`})`,
        }}
        className={style.head}
      >
        <div className={style.input}>
          <Input onSubmit={addNewTodo} />
        </div>
        <div className={style.commandPanel}>
          <CommandPanel
            todos={todos}
            onFilterChanged={value => setFilter(value)}
            onRemoveCompleted={removeCompleted}
          />
        </div>
      </div>
      <div className={style.body}>
        <List
          todos={todos}
          filter={filter}
          onDone={todoDone}
          onRemove={todoRemove}
        />
      </div>
    </div>
  );
}
