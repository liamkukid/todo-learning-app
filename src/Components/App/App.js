import NewTodoForm from '../NewTodoForm/NewTodoForm';
import List from '../TodosList/List';
import CommandPanel from '../CommandPanel/CommandPanel';
import style from './App.module.scss';

export default function App() {
  return (
    <div className={style.app}>
      <div
        style={{
          backgroundImage: `url(${`${process.env.PUBLIC_URL}/work_desk.jpg`})`,
        }}
        className={style.head}
      >
        <div className={style.newTodoForm}>
          <NewTodoForm />
        </div>
        <div className={style.commandPanel}>
          <CommandPanel />
        </div>
      </div>
      <div className={style.body}>
        <List />
      </div>
    </div>
  );
}
