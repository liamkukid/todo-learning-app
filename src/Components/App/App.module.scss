@import '../../styles/variables.scss';

.app {
  display: grid;
  gap: 2px;
  width: 100vw;
  height: 100vh;
  grid-template-rows: 3fr 4fr;
  background-color: var(--navy);
}

.head {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.newTodoForm {
  height: 90%;
  display: flex;
  align-items: center;  
}

.commandPanel {
  height: 10%;
  display: flex;
  align-items: center;  
}

.body {
  background-color: var(--ligth-beige);  
  overflow: overlay;
}