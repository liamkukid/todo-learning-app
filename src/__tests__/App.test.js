import renderer from 'react-test-renderer';
import App from '../App';

jest.mock('react-redux');

it('renders correctly for App', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
