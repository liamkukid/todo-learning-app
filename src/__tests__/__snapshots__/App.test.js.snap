// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for App 1`] = `
<div
  className="app"
>
  <div
    className="head"
    style={
      Object {
        "backgroundImage": "url(/work_desk.jpg)",
      }
    }
  >
    <div
      className="input"
    >
      <form
        className="form_content"
        onSubmit={[Function]}
      >
        <input
          className="form_input"
          id="todo_input"
          name="todo_input"
          placeholder="New todo..."
          type="text"
        />
        <button
          className="form_submit"
          type="submit"
        >
          Save
        </button>
      </form>
    </div>
    <div
      className="commandPanel"
    >
      <div
        className="panel"
      >
        <span
          className="span_items_left"
        >
          no
           items left
        </span>
        <button
          className="button"
          onClick={[Function]}
          type="button"
          value="All"
        >
          All
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
          value="Active"
        >
          Active
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
          value="Completed"
        >
          Completed
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
          value="Remove Completed"
        >
          Remove Completed
        </button>
      </div>
    </div>
  </div>
  <div
    className="body"
  >
    <div />
  </div>
</div>
`;
